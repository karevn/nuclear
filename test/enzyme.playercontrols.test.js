import React from 'react';
import { shallow, mount } from 'enzyme';
import FontAwesome from 'react-fontawesome';

import PlayerControls from '../app/components/PlayerControls'
import PlayPauseButton from '../app/components/PlayerControls/PlayPauseButton';
import PlayerButton from '../app/components/PlayerControls/PlayerButton';


describe('Rendering <PlayerControls /> and its children', () => {
  it('Should render the PlayPauseButton and two PlayerButton ', () => {
    const wrapper = shallow(<PlayerControls />);
    expect(wrapper.containsMatchingElement(<PlayPauseButton />)).toBe(true);
    expect(wrapper.find(PlayerButton)).toHaveLength(2);
    expect(wrapper.find(PlayerButton).first().props().icon).toBe('step-backward');
    expect(wrapper.find(PlayerButton).last().props().icon).toBe('step-forward');
  });
});

describe('Back button onClick', () => {
  it('has an onclick event that gets triggered', () => {
    const spy = jest.fn();
    const wrapper = shallow(<PlayerControls back={spy} />);
    wrapper.find(PlayerButton).first().simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});

describe('Forward button onClick', () => {
  it('has an onclick event that gets triggered', () => {
    const spy = jest.fn();
    const wrapper = shallow(<PlayerControls forward={spy} />);
    wrapper.find(PlayerButton).last().simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});

describe('<PlayPauseButton/> loading', () => {
  it('Loading indicator showing', () => {
    const wrapper = mount(<PlayPauseButton loading={true} playing={false} onClick={undefined} />);
    expect(wrapper.find(FontAwesome).prop('name')).toBe('spinner');
  });
});

describe('<PlayPauseButton/> playing', () => {
  it('Shows pause button', () => {
    const wrapper = mount(<PlayPauseButton loading={false} playing={true} onClick={undefined} />);
    expect(wrapper.find(FontAwesome).prop('name')).toBe('pause');
  });
});

describe('<PlayPauseButton/> ready to play', () => {
  it('Shows play button', () => {
    const wrapper = mount(<PlayPauseButton loading={false} playing={false} onClick={undefined} />);
    expect(wrapper.find(FontAwesome).prop('name')).toEqual('play');
  });
});

describe('<PlayPauseButton/> onClick', () => {
  it('Shows play button', () => {
    const spy = jest.fn();
    const wrapper = mount(<PlayPauseButton loading={false} playing={false} onClick={spy} />);
    wrapper.find('a').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
