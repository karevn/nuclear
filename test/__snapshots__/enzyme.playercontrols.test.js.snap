// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayPauseButton/> loading Loading indicator showing 1`] = `
<withI18nextTranslation(PlayPauseButton)
  loading={true}
  playing={false}
>
  <PlayPauseButton
    i18n={Object {}}
    loading={true}
    playing={false}
    t={[Function]}
    tReady={true}
  >
    <div
      aria-label="play-pause-button"
      className="play_pause_button_container loading player_button_disabled"
    >
      <a
        href="#"
      >
        <FontAwesome
          name="spinner"
          pulse={true}
        >
          <span
            aria-hidden={true}
            className="fa fa-spinner fa-pulse"
          />
        </FontAwesome>
      </a>
    </div>
  </PlayPauseButton>
</withI18nextTranslation(PlayPauseButton)>
`;

exports[`<PlayPauseButton/> onClick Shows play button 1`] = `
<withI18nextTranslation(PlayPauseButton)
  loading={false}
  onClick={[MockFunction]}
  playing={false}
>
  <PlayPauseButton
    i18n={Object {}}
    loading={false}
    onClick={[MockFunction]}
    playing={false}
    t={[Function]}
    tReady={true}
  >
    <div
      aria-label="play-pause-button"
      className="play_pause_button_container"
    >
      <a
        href="#"
        onClick={[MockFunction]}
      >
        <FontAwesome
          name="play"
        >
          <span
            aria-hidden={true}
            className="fa fa-play"
          />
        </FontAwesome>
      </a>
    </div>
  </PlayPauseButton>
</withI18nextTranslation(PlayPauseButton)>
`;

exports[`<PlayPauseButton/> playing Shows pause button 1`] = `
<withI18nextTranslation(PlayPauseButton)
  loading={false}
  playing={true}
>
  <PlayPauseButton
    i18n={Object {}}
    loading={false}
    playing={true}
    t={[Function]}
    tReady={true}
  >
    <div
      aria-label="play-pause-button"
      className="play_pause_button_container player_button_disabled"
    >
      <a
        href="#"
      >
        <FontAwesome
          name="pause"
        >
          <span
            aria-hidden={true}
            className="fa fa-pause"
          />
        </FontAwesome>
      </a>
    </div>
  </PlayPauseButton>
</withI18nextTranslation(PlayPauseButton)>
`;

exports[`<PlayPauseButton/> ready to play Shows play button 1`] = `
<withI18nextTranslation(PlayPauseButton)
  loading={false}
  playing={false}
>
  <PlayPauseButton
    i18n={Object {}}
    loading={false}
    playing={false}
    t={[Function]}
    tReady={true}
  >
    <div
      aria-label="play-pause-button"
      className="play_pause_button_container player_button_disabled"
    >
      <a
        href="#"
      >
        <FontAwesome
          name="play"
        >
          <span
            aria-hidden={true}
            className="fa fa-play"
          />
        </FontAwesome>
      </a>
    </div>
  </PlayPauseButton>
</withI18nextTranslation(PlayPauseButton)>
`;

exports[`Back button onClick has an onclick event that gets triggered 1`] = `
<div
  className="player_controls_container"
>
  <PlayerButton
    ariaLabel="previous-button"
    icon="step-backward"
    onClick={[MockFunction]}
  />
  <withI18nextTranslation(PlayPauseButton) />
  <PlayerButton
    ariaLabel="next-button"
    icon="step-forward"
  />
</div>
`;

exports[`Forward button onClick has an onclick event that gets triggered 1`] = `
<div
  className="player_controls_container"
>
  <PlayerButton
    ariaLabel="previous-button"
    icon="step-backward"
  />
  <withI18nextTranslation(PlayPauseButton) />
  <PlayerButton
    ariaLabel="next-button"
    icon="step-forward"
    onClick={[MockFunction]}
  />
</div>
`;
